addons/website_forum/models/forum_post.py

```diff
@@ -276,27 +283,22 @@ def _search_can_view(self, operator, value):
 
         user = self.env.user
         # Won't impact sitemap, search() in converter is forced as public user
-        if self.env.is_admin():
-            return [(1, '=', 1)]
-
-        req = """
-            SELECT p.id
-            FROM forum_post p
-                   LEFT JOIN res_users u ON p.create_uid = u.id
-                   LEFT JOIN forum_forum f ON p.forum_id = f.id
-            WHERE
-                (p.create_uid = %s and f.karma_close_own <= %s)
-                or (p.create_uid != %s and f.karma_close_all <= %s)
-                or (
-                    u.karma > 0
-                    and (p.active or p.create_uid = %s)
-                )
-        """
-
-        op = 'inselect' if operator == '=' else "not inselect"
+        # not env.is_admin as this field is used in ir.rules
+        all_access_groups = ','.join(self._get_can_view_all_access_groups())
+        if user._is_admin() or all_access_groups and user.user_has_groups(all_access_groups):
```

Comment:
can we just return `base.group_erp_manager` by default in `_get_can_view_all_access_groups` ?